### DISCUSSION TITLE
Legal GDPR Cookie Banner (Fully customizable and with autoblocking)

### DISCUSSION THREAD

**Discussion by MattiasBaldi** [elementor/elementor#23344] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I'm trying to find a legal cookie banner that blocks cookies from third party when clicking reject and also doesn't place cookies on my device before consent has given (prior consent). 

For this there a few options. Use a free but very limited design-friendly template from sites like ibuenda, cookieinformation.com etc. often with a bunch of restrictions, that allow for very limited design and are still illegal to use. Or straight up, dont read the rules of GDPR and just display a banner that has no legal functionality like 90% of sites out there.

These are the actual rules and the checklist of GDPR concerning Cookies, right now 90% only follow the first step: 

- Ensure that your website has a cookie pop-up (cookie banner).
- Collect and store consents from your visitors.
- Provide visitors the option to decline cookies (opt-out).
- Withhold cookies until you have received consent (prior consent).
- Describe all cookies, their purposes, and lifetimes (information obligation).
- Maintain a comprehensive list of all third-party cookies that receive data about your visitors.

### Describe the solution you'd like

It would be nice if there was a banner widget or pop-up that is premade, and is fully customizable. 

It should provide opportunies to autoblocking of the most popular third party cookies like google analytics, tag manager, ads, facebook pixel (now meta pixel), tiktok pixel, hotjar etc.

It should store consent the same way submissions are stored from contact forms inside of elementor

It should work like a pop-up feature that also allows for full customization like pop-ups does already... so you are not restricted to say, certain border radius, fonts or colors etc. 

Your audience care about simplicity and design!

**Comment by MattiasBaldi** [comment]
I tried these;

Websites all with dashboards from external webapps; 

https://www.iubenda.com/da/ (Not customizable except for a few colors, )
https://cookieinformation.com (not customizable, limited visitors)
https://securiti.ai/blog/cookie-banner/ (semi customizable)
https://www.cookiebot.com/en/cookie-banner/ (not customizable, no consent log, limited visitors)
https://cookie-script.com/ (not customizable, no consent log, limited visitors)

Plugins; 

https://wordpress.org/plugins/cookie-consent-autoblock/ (Seems to be outdated autoblocking doesn't work)
https://wordpress.org/plugins/cookie-notice/ (Semi customizable, limited visitors, external web app)
https://wordpress.org/plugins/cookie-law-info/ (basic customization for free subs, external web app, limited visitors
https://wordpress.org/plugins/beautiful-and-responsive-cookie-consent/ (no autoblocking, not GDPR compliant)

**The rest of the plugins are semi expensive pr. year payments. Varying between 40-80 euros** 

Alternatives was to code my own, which was from the looks of it not very hard, but I'm not a pro developer and I reckon a lot more people could have use of this..

**Reply by MattiasBaldi** [reply]
And then there is A LOT of youtube tutorials (also for elementor) on how to make a cookie banner, but none of those were actually compliant with GDPR. They did not provide a reject button, nor did they "Withhold cookies until you have received consent (prior consent)."

So simply it's just a banner. I think the GDPR regulations are rather complicated for most people, so the reality is, that there is a lot of websites that run with illegal cookie banners. Most are unaware of this, and are smaller companies, perfectly fit to be elementor pro clients. I really hope someone makes this, it would make life so much more simple for the most of us!

**Reply by FannyLuxuryBrands** [reply]
Completely agree with you, I hope something will come out!!
